File: solutions/scala/raindrops/1/src/main/scala/Raindrops.scala
object Raindrops {
  def convert(n: Int): String = {
  	val res = 
	  	(1 to n)
	  	.filter(n % _ == 0)
	  	.foldLeft("") { case (acc, x) =>
	  		x match {
	  			case 3 => acc ++ "Pling"
	  			case 5 => acc ++ "Plang"
	  			case 7 => acc ++ "Plong"
	  			case _ => acc
	  		}
	  	}

	  res.isEmpty match {
	  	case true  => n.toString
	  	case false => res
	  }
  }
}

File: solutions/go/raindrops/1/raindrops.go
package raindrops

import "fmt"

func Convert(number int) string {
	str := ""

	if number%3 == 0 {
		str += "Pling"
	}

	if number%5 == 0 {
		str += "Plang"
	}

	if number%7 == 0 {
		str += "Plong"
	}

	if len(str) == 0 {
		return fmt.Sprintf("%d", number)
	} else {
		return str
	}
}
